File: PHP/Film.php
<?php
require_once __DIR__ . '/Konten.php';

class Film extends Konten {
    protected $durasi;
    protected $sutradara;
    protected $tahunRilis;

    // Konstruktor ini membuat objek Film dengan data tambahan khusus film.
    public function __construct($id = 0, $judul = "", $genre = "", $durasi = 0, $sutradara = "", $tahunRilis = 0) {
        parent::__construct($id, $judul, $genre);
        $this->durasi = (int)$durasi;
        $this->sutradara = $sutradara;
        $this->tahunRilis = (int)$tahunRilis;
    }

    // Method ini mengembalikan durasi film dalam menit.
    public function getDurasi(){
        return $this->durasi; 
    }

    // Method ini mengubah durasi film.
    public function setDurasi($durasi){
        $this->durasi = (int)$durasi; 
    }

    // Method ini mengembalikan nama sutradara film.
    public function getSutradara(){
        return $this->sutradara; 
    }

    // Method ini mengubah nama sutradara film.
    public function setSutradara($sutradara){
        $this->sutradara = $sutradara; 
    }

    // Method ini mengembalikan tahun rilis film.
    public function getTahunRilis(){
        return $this->tahunRilis; 
    }

    // Method ini mengubah tahun rilis film.
    public function setTahunRilis($tahunRilis){
        $this->tahunRilis = (int)$tahunRilis; 
    }

    // Method ini menampilkan data Konten dan data khusus film.
    public function tampilkanData(){
        parent::tampilkanData();
        echo "Durasi : " . $this->durasi . " menit<br>";
        echo "Sutradara : " . $this->sutradara . "<br>";
        echo "Tahun Rilis : " . $this->tahunRilis . "<br>";
    }
}
?>

File: Python/Film.py
from Konten import Konten

class Film(Konten):
    # Konstruktor ini membuat objek Film dengan data tambahan khusus film.
    def __init__(self, id=0, judul="", genre="", durasi=0, sutradara="", tahun_rilis=0):
        super().__init__(id, judul, genre)
        self.__durasi = durasi
        self.__sutradara = sutradara
        self.__tahun_rilis = tahun_rilis

    # Method ini mengembalikan durasi film dalam menit.
    def getDurasi(self):
        return self.__durasi

    # Method ini mengubah durasi film.
    def setDurasi(self, durasi):
        self.__durasi = durasi

    # Method ini mengembalikan nama sutradara film.
    def getSutradara(self):
        return self.__sutradara

    # Method ini mengubah nama sutradara film.
    def setSutradara(self, sutradara):
        self.__sutradara = sutradara

    # Method ini mengembalikan tahun rilis film.
    def getTahunRilis(self):
        return self.__tahun_rilis

    # Method ini mengubah tahun rilis film.
    def setTahunRilis(self, tahun_rilis):
        self.__tahun_rilis = tahun_rilis

    # Method ini menampilkan data Konten dan data khusus film.
    def tampilkanData(self):
        super().tampilkanData()
        print(f"Durasi : {self.__durasi} menit")
        print(f"Sutradara : {self.__sutradara}")
        print(f"Tahun Rilis : {self.__tahun_rilis}")
